### PR TITLE
fix: load vs-code webview resources from local host instead of getting them from CDN

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?
If configured, forwards requests to the local domain instead of sending them to the CDN.

To turn the redirection on, the `WEBVIEW_LOCAL_RESOURCES` environment variable should be set into the devfile (or in the editor devfile)
```
      env:
        - name: WEBVIEW_LOCAL_RESOURCES
          value: 'yes'
```

If configured, those four types of the requests / resources will be redirected to the same, from which vs code main frame was loaded

- resources that initialize the webview and the webworker ( with HTTP scheme )
  `${WebviewResourcesBaseURI}oss-dev/static/out/vs/workbench/contrib/webview/browser/pre`

- resources from a built-in extension ( FILE scheme )
  `file:///checode/checode-linux-libc/extensions/markdown-language-features/media/pre.js`

- resources, that are located in the current project ( FILE scheme )
  `file:///projects/web-nodejs-sample/images/default.png`

- resources from installed extension ( FILE scheme )
  `file:///checode/remote/extensions/atlassian.atlascode-3.0.2-universal`

`entrypoint-volume.sh` script has instructions to create symbolic links to `/project` and to `/checode/remote/extensions`. It is needed to load the resources, that are located in the current project and in the installed extensions [ 3 and 4 ].

Screenshot from 2023-04-05 19-10-28

### What issues does this PR fix?

https://issues.redhat.com/browse/CRW-3885
https://issues.redhat.com/browse/CRW-4178

#### Webview resources are local-workspace based:

1. Create a workspace from the repository https://github.com/vitaliy-guliy/web-nodejs-sample/tree/test-webviews-local-resources ( it will use prebuilt [quay.io/vgulyy/che-code:webviews-behind-a-proxy-stable](https://quay.io/vgulyy/che-code:webviews-behind-a-proxy-stable) editor image )

2. Open README.md at the project root, click Open Preview at the right top
3. Install Jira and Bitbucket extension, open Atlassian Settings tab ( webview based )

Before 2 and 3 steps, open the browser console, navigate to Network.
After 2 and 3 check there is no requests to vscode-cdn.net.

#### Webview resources are CDN-based:

The same as above, but create a workspace from https://github.com/vitaliy-guliy/web-nodejs-sample/tree/test-webviews-resources-on-cdn
Check the browser console, all the webview resources must be loaded from [vscode-cdn.net](vscode-cdn.net)

### CODE REVIEW

**Comment by vitaliy-guliy** [comment]
Discussing with @azatsarynnyy another approach, but for that we need so sniff a network traffic a bit.
We think that if the local resource is requested ( at the screenshot - all the request below the selection ), the browse does not send the requests to `vscode-cdn.net` at all. Instead of that the request could caught by webworker (not sure here) under the hood and be handled in another way

Screenshot from 2023-04-12 17-29-28

File: code/src/vs/workbench/contrib/webview/common/che/webviewLocalResources.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/**********************************************************************
+ * Copyright (c) 2022 Red Hat, Inc.
```
Comment (RomanNikitenko) [inline code/src/vs/workbench/contrib/webview/common/che/webviewLocalResources.ts]: 2023?

File: code/src/vs/workbench/contrib/webview/browser/webviewElement.ts [hunk 2]
```diff
@@ -37,6 +37,7 @@ import { WebviewFindDelegate, WebviewFindWidget } from 'vs/workbench/contrib/web
 import { FromWebviewMessage, KeyEvent, ToWebviewMessage } from 'vs/workbench/contrib/webview/browser/webviewMessages';
 import { decodeAuthority, webviewGenericCspSource, webviewRootResourceAuthority } from 'vs/workbench/contrib/webview/common/webview';
 import { IWorkbenchEnvironmentService } from 'vs/workbench/services/environment/common/environmentService';
+import { alternativeWebviewExternalEndpoint, loadWebviewResourcesFromLocalWorkspace } from '../common/che/webviewLocalResources';
```
Comment (RomanNikitenko) [inline code/src/vs/workbench/contrib/webview/browser/webviewElement.ts]: `from 'vs/workbench/contrib/webview/common/che/webviewLocalResources';`

File: code/src/vs/workbench/contrib/webview/common/webview.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import { CharCode } from 'vs/base/common/charCode';
 import { Schemas } from 'vs/base/common/network';
 import { URI } from 'vs/base/common/uri';
+import { redirectToLocalWorkspace } from './che/webviewLocalResources';
```
Comment (RomanNikitenko) [inline code/src/vs/workbench/contrib/webview/common/webview.ts]: `import { redirectToLocalWorkspace } from 'vs/workbench/contrib/webview/common/che/webviewLocalResources';`

**Comment by vitaliy-guliy** [comment]
Superseded by https://github.com/che-incubator/che-code/pull/211

**Pull request closed without merging**